BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/location/distance/DistanceUtil.java

```diff
@@ -21,4 +27,19 @@ public static int getMaximumEvenIncrement(double distance) {
         return (int) Math.pow(10, Math.floor(Math.log10(distance)));
     }
 
+    public static Location walkingSpeedFilter(Location oldLocation, Location newLocation) {
+        return speedFilter(oldLocation, newLocation, HUMAN_WALKING_SPEED);
+    }
+
+    public static Location speedFilter(Location oldLocation, Location newLocation, double metersPerSecond) {
+        double distance = oldLocation.getDistanceTo(newLocation);
+        double angle = oldLocation.getAngleTo(newLocation);
+        long timestampDelta = newLocation.getTimestamp() - oldLocation.getTimestamp();
+        metersPerSecond = metersPerSecond * ((float) timestampDelta / TimeUnit.SECONDS.toMillis(1));
```

Comment:
You overwrite the original maximum speed here. Calculate the speed between the locations instead and check if it is faster than the maximum speed.